R/convert-from-datacamp.R

```diff
@@ -25,82 +25,65 @@ datacamp_to_learnr_pkg <-
 
     # options(usethis.quiet = TRUE)
 
-    if (is.null(cur_path)) {
-      cur_path = getwd()
+    if (is.null(dc_path)) {
+      dc_path = getwd()
     }
     #
     # # For accidental sourcing... -_-
     # stop()
     #
     # # Checking if project is under git and if changes are not committed
-    if (!in_repository(cur_path))
+    if (!in_repository(dc_path))
       stop("Please make that the files are under Git version control.")
-    if (length(status(cur_path)$unstaged) != 0)
+    if (length(status(dc_path)$unstaged) != 0)
       stop("Please make sure to commit changes before running.")
     #
     # # Listing all files -------------------------------------------------------
 
-    ## needs to check to see if there are dashes in the folder name
-
-    pkgname <- basename(cur_path)
-    if (grepl("-", pkgname)) {
-      if (is.null(rename)) {
-        pkgname <- gsub("-", "", pkgname)
-      } else {
-        pkgname <- rename
-      }
-
-      new_path <- strsplit(cur_path, "/")[[1]]
-      new_path <- new_path[-length(new_path)]
-      # new_path <- new_path[-1]
-      new_path <- paste(new_path, collapse = .Platform$file.sep)
-      new_path <- file.path(new_path, pkgname)
-      file.rename(cur_path, new_path)
-      cur_path <- new_path
-    }
+
+    pkgname <- basename(lrnr_pkg_path)
     pkgname <- paste(pkgname, ".Rproj", sep = "")
 
     all_files <-
-      fs::dir_ls(cur_path,
+      fs::dir_ls(dc_path,
              recurse = TRUE,
              type = "file",
              all = TRUE)
-    proj_name <- str_subset(all_files, "\\.Rproj$")
 
 
 
-    #### No RPoject, which would seem to be default??
-    if (length(proj_name) == 0) {
-      tmp <- paste(pkgname, ".Rproj", sep = "")
-      proj_name = file.path(cur_path, tmp)
-    }
     # # Create as package -------------------------------------------------------
-
-    usethis::create_package(fs::path_dir(proj_name))
+    ### this will fail if there are dashes, in the name, could provide an earlier error
```

Comment:
Yes, you are right. I was going to implement a simple check and stop, something like:

```
if (str_detect(pkgname, "( |-|_)")) stop("The new learnr package shouldn't have any spaces, dashes or underscores")
```

You wanna insert something like that in?